datafusion/optimizer/src/simplify_expressions/simplify_exprs.rs

```diff
@@ -123,10 +123,14 @@ impl SimplifyExpressions {
         let name_preserver = NamePreserver::new(&plan);
         let mut rewrite_expr = |expr: Expr| {
             let name = name_preserver.save(&expr);
+            let original = expr.clone();
```

Comment:
the extra clone here worries me. 

It seems like maybe a better approach would be to change `Simplifier::simplify` to return `Transformed`?

Reply:
🤦  -- it seems that is what @jayzhan211  suggested above in https://github.com/apache/datafusion/pull/15735#issuecomment-2809315034